src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Authorization/AddAuthorizationBuilderFixer.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Analyzers.Authorization.Fixers;
+
+[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+public sealed class AddAuthorizationBuilderFixer : CodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticDescriptors.UseAddAuthorizationBuilder.Id);
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        if (root == null)
+        {
+            return;
+        }
+
+        var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+        if (semanticModel == null)
+        {
+            return;
+        }
+
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            if (CanReplaceWithAddAuthorizationBuilder(diagnostic, root, out var invocation))
+            {
+                const string title = "Use 'AddAuthorizationBuilder'";
+                context.RegisterCodeFix(
+                    CodeAction.Create(title,
+                        cancellationToken => ReplaceWithAddAuthorizationBuilder(diagnostic, root, context.Document, invocation),
+                        equivalenceKey: title),
```

Comment:
```suggestion
                        equivalenceKey: EquivalencyKey.Create(DiagnosticDescriptors.UseAddAuthorizationBuilder.Id)),
```

Reply:
Hmm, I'm not seeing anything for this `EquivalencyKey` class. Where would I be able to find this?

Reply:
Total brain fart on my part. I think we can just do `equivalenceKey : DiagnosticDescriptors.UseAddAuthorizationBuilder.Id`.

I notice that there are some places in our code where we use the title as the equivalence key and others where we use the diagnostic ID, so it appears we're inconsistent in our approach here.

Based on the docs [here](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md#selecting-an-equivalence-key-for-code-actions), using the title is recommended for most cases unless you have a fixer that targets different values.

TL;DR: Disregard my feedback here!

Reply:
Good to know! I've updated this to use the DiagnosticDescriptor Id as the equivalenceKey.